hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1118,15 +1118,28 @@ public AbfsInputStream openFileForRead(final Path path,
 
       String relativePath = getRelativePath(path);
 
-      final AbfsRestOperation op = client
-          .getPathStatus(relativePath, false, tracingContext);
+      AbfsRestOperation op;
+      if (OperativeEndpoint.isReadEnabledOnDFS(getPrefixMode(), abfsConfiguration)) {
```

Comment:
Here we should not fallback. Fallback is not intended for system and user properties call. We should add a handling for implicit directory here. Have added similar checks in mkdir PR for reference.

Reply:
Also we should have separate fallback for metadata call, not dependent on the read fallback.

Reply:
Taken